pyiron_base/job/path.py

```diff
@@ -473,7 +473,7 @@ def __init__(self, db, job_id=None, db_entry=None, user=None):
             job_path = db_entry["projectpath"]
         else:
             job_path = ''
-        job_path += db_entry["project"] + hdf5_file + db_entry["subjob"]
+        job_path = os.path.join(job_path, db_entry["project"], hdf5_file) + db_entry["subjob"]
```

Comment:
As far as I understand this is the line that cause the bug of not being able to load the jobs, correct? If this is the case can you tell me what is the input and what is the output? In case there is a missing "/" in the database entry, then our search does not work, so this might be the wrong place to fix it.

Reply:
The problem in this line is that the `db_entry["project"]` is not terminated with a '/'.  I also tried to ensure it is added [here](https://github.com/pyiron/pyiron_base/blob/38e3f0c4ace3a7d939ef3b4527c2f3741e082d8b/pyiron_base/archiving/import_archive.py#L97) before the job is added to the database, but it still got swallowed somewhere.  I did not investigate the details though.

I would rather keep this change, because it makes the relevant code more robust instead of trying to ensure we always provide correct input.

Reply:
The issue is that in the database we have the option to search in the current project and in all sub projects and as far as I remember this is realised by having the same name e.g. `/project/name/` in both the project and the sub project.

Reply:
Here is the fix just for the job reload https://github.com/pyiron/pyiron_base/pull/178/

Reply:
It seems to me the old code would still break in that case, so if it is an option than it would have to take a different code path.  Since `os.path.join` only ensures there's '/' between the arguments and nothing more, I don't see how it could interfere with this.  Anyways #178 seems to work, so it is ok for now.

Reply:
I agree the code breaks and I am open to merge this change, but we should be careful to have all `project` entries in the database end with `/` otherwise the database search does not work - at least that is what I remember. There was an issue with using `/` in database queries.